src/cli/format-results-cache.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+// Inspired by LintResultsCache from ESLint
+// https://github.com/eslint/eslint/blob/c2d0a830754b6099a3325e6d3348c3ba983a677a/lib/cli-engine/lint-result-cache.js
+
+const fileEntryCache = require("file-entry-cache");
+const stringify = require("fast-json-stable-stringify");
+// eslint-disable-next-line no-restricted-modules
+const { version: prettierVersion } = require("../index.js");
+const { createHash } = require("./utils.js");
+
+const optionsHashCache = new WeakMap();
+const nodeVersion = process && process.version;
+
+/**
+ * @param {*} options
+ * @returns {string}
+ */
+function getHashOfOptions(options) {
+  if (optionsHashCache.has(options)) {
+    return optionsHashCache.get(options);
+  }
+  const hash = createHash(
+    `${prettierVersion}_${nodeVersion}_${stringify(options)}`
+  );
+  optionsHashCache.set(options, hash);
+  return hash;
+}
+
+/**
+ * @typedef {{ hashOfOptions?: string }} OurMeta
+ * @typedef {import("file-entry-cache").FileDescriptor} FileDescriptor
+ *
+ * @param {import("file-entry-cache").FileDescriptor} fileDescriptor
+ * @returns {FileDescriptor["meta"] & OurMeta}
+ */
+function getMetadataFromFileDescriptor(fileDescriptor) {
+  return fileDescriptor.meta;
+}
+
+const cacheStrategies = ["metadata", "content"];
+const cacheStrategiesSet = new Set(cacheStrategies);
+
+/**
+ * @typedef {"metadata" | "content"} CacheStrategy
+ *
+ * @param {string} value
+ * @returns {value is CacheStrategy}
+ */
+function isValidCacheStrategy(value) {
+  return cacheStrategiesSet.has(value);
+}
+
+class FormatResultsCache {
+  /**
+   * @param {string} cacheFileLocation The path of cache file location. (default: `node_modules/.cache/prettier/prettier-cache`)
+   * @param {string} cacheStrategy
+   */
+  constructor(cacheFileLocation, cacheStrategy) {
+    if (!isValidCacheStrategy(cacheStrategy)) {
+      const errorMessage = `Cache strategy must be one of: ${cacheStrategies
+        .map((strategy) => `"${strategy}"`)
+        .join(", ")}`;
+      throw new Error(errorMessage);
```

Comment:
Maybe this can be done in flag validation?

Reply:
[9499918](https://github.com/prettier/prettier/pull/12800/commits/94999181aeba4c574c02a1cbf0f8a3504705c97a) :+1: